src/wallet/export.rs

```diff
@@ -142,12 +147,24 @@ impl WalletExport {
             blockheight,
         };
 
-        let desc_to_string = |d: &Descriptor<DescriptorPublicKey>| {
-            let descriptor =
-                d.to_string_with_secret(&wallet.change_signers.as_key_map(wallet.secp_ctx()));
-            remove_checksum(descriptor)
+        let change_descriptor = match wallet
+            .public_descriptor(KeychainKind::Internal)
+            .map_err(|_| "Invalid change descriptor")?
+            .is_some()
+        {
+            false => None,
+            true => {
+                let descriptor = wallet
+                    .get_descriptor_for_keychain(KeychainKind::Internal)
+                    .to_string_with_secret(
+                        &wallet
+                            .get_signers(KeychainKind::Internal)
+                            .as_key_map(wallet.secp_ctx()),
+                    );
+                Some(remove_checksum(descriptor))
+            }
         };
-        if export.change_descriptor() != wallet.change_descriptor.as_ref().map(desc_to_string) {
+        if export.change_descriptor() != change_descriptor {
```

Comment:
Yes, unfortunately this export format only supports descriptors that match this specific structure. See [the spec here](https://github.com/Fonta1n3/FullyNoded/blob/10b7808c8b929b171cca537fb50522d015168ac9/Docs/Wallets/Wallet-Export-Spec.md#change-keys)

Reply:
Ya the spec does makes sense.. And its also nasty to generalize it.. 
I am just thinking on what happens when someone doesn't respect the the spec at wallet creation, and then finds it out at the export.. We do not seem to be enforcing this at `wallet::new`..

Reply:
I don't think we should enforce this in the wallet constructor, it would be very limiting. I think it's fine that some descriptors can't be exported using this protocol, this is meant to be used to share descriptors with FullyNoded specifically and if you don't care about it you should not be limited in the set of descriptors you can use.

Maybe we should make it more clear that this is a FullyNoded-specific export, not a "generic" one. And we could design a different spec for arbitrary descriptors ourselves, or see if there's something else out there that we can reuse.

Reply:
> Maybe we should make it more clear that this is a FullyNoded-specific export, not a "generic" one.

Ya that makes sense.. I am just worried about someone not noticing this and get into trouble later. Ya explicitly clarifying it would be helpful. And we can figure the generic way later on..